source/slang/slang-check-expr.cpp

```diff
@@ -1324,9 +1324,11 @@ namespace Slang
                             // of explicit default initializers for `struct` fields to
                             // make this a major concern (since they aren't supported in HLSL).
                             //
-                            RefPtr<InitializerListExpr> initListExpr = m_astBuilder->create<InitializerListExpr>();
+                            InitializerListExpr* initListExpr = m_astBuilder->create<InitializerListExpr>();
                             auto checkedInitListExpr = visitInitializerListExpr(initListExpr);
-                            return coerce(typeExp.type, initListExpr);
+
+                            // TODO(JS): I changed this to checkInitListExpr form initListExpr
```

Comment:
Your change here is more correct than the original code, which probably worked by accident, since `visitInitializerListExpr` would always return the same AST node that was passed in.